library/core/src/iter/traits/iterator.rs

```diff
@@ -2216,6 +2216,73 @@ pub trait Iterator {
         Some(self.fold(first, f))
     }
 
+    /// Reduces the elements to a single one by repeatedly applying a reducing operation. If the
+    /// closure returns a failure, the failure is propagated back to the caller immediately.
+    ///
+    /// The return type of this method depends on the return type of the closure. If the closure
+    /// returns `Result<Self::Item, E>`, then this function will return `Result<Option<Self::Item>,
+    /// E>`. If the closure returns `Option<Self::Item>`, then this function will return
+    /// `Option<Option<Self::Item>>`.
+    ///
+    /// When called on an empty iterator, this function will return either `Some(None)` or
+    /// `Ok(None)` depending on the type of the provided closure.
+    ///
+    /// For iterators with at least one element, this is essentially the same as calling
+    /// [`try_fold()`] with the first element of the iterator as the initial accumulator value.
+    ///
+    /// [`try_fold()`]: Iterator::try_fold
+    ///
+    /// # Examples
```

Comment:
Could you please add a `Result`-using example here too?  I think that would help users see that it'll give a `Result<Option<_>, _>` -- understanding that from the expanded `ChangeOutputType` is non-trivial in the rustdoc.

Reply:
Sure.

Reply:
Oh, I had missed the section you had in the text about the return type.  With that and the `Result` use in the tests, I'd be fine with it as-is too.  Let me know which you'd prefer.

Reply:
At your suggestion, I would like to have an additional example in the documentation making clear that `Result` can be used as well. I think the documentation is currently clear but there's no harm in having an additional example.

Reply:
I've added one of the tests to the documentation. Is that acceptable?

Reply:
Looks good; thanks!